### ISSUE TITLE
Support both UPPER and lower case versions of proxy environment variables

### ISSUE THREAD

**Issue by xphyr** [gomods/athens#1192]
I have been trying to figure out why Athens was not working behind our proxy and finally figured it out.  On our hosts we define "http_proxy", "https_proxy" and "no_proxy", not "HTTP_PROXY", "HTTPS_PROXY" and "NO_PROXY".  I was able to find that this traces back to this here:

https://github.com/gomods/athens/blob/master/pkg/module/go_get_fetcher.go#L165-L167

NOTE:  The go toolchain actually supports both UPPER and lower case versions of these environment variables.  The go language also supports both UPPER and lower case versions of these environment variables as well for proxy support.

**To Reproduce**

export http_proxy=<your proxy here>
export https_proxy=<your proxy here>
export no_proxy=<list of domains to not proxy>
./athens

Athens will not be able to retrieve any code as the underlying go tool will be unable to retrieve anything via the proxy.

**Expected behavior**
Athens should support both upper and lower case versions of these proxy specific environment variables.

**Environment (please complete the following information):**
 - OS: Linux
 - Go version : 1.12.x
 - Proxy version : 0.2
 - Storage (fs/mongodb/s3 etc.) : fs

**Comment by arschles** [comment]
This was closed by #1192. Closing

**Comment by ulrichSchreiner** [comment]
hi,

the 0.4.0 version of athens does not work in our environment any more. we set `HTTP_PROXY` and `HTTPS_PROXY` but not the values with lowercase. i think your fix passes the environment variables with uppercase (which have values in our env) and with lowercase (which do not have values in our env) to your subprocess environment. now the environment contains
~~~
HTTP_PROXY=http://...
HTTPS_PROXY=http://...
http_proxy=
https_proxy=
~~~

and now our instance of athens cannot access the internet any more!

i think you should check if an env-variable is defined and only then pass this to your environment.
